lms/djangoapps/mobile_api/middleware.py

```diff
@@ -0,0 +1,124 @@
+            latest_version = request_cache_dict[self.LATEST_VERSION_HEADER]
+            user_app_version = request_cache_dict[self.USER_APP_VERSION]
+            if (latest_version != self.NO_LATEST_VERSION and
+                    parsed_version(user_app_version) < parsed_version(latest_version)):
+                response[self.LATEST_VERSION_HEADER] = latest_version
+        return response
+
+    def _get_cache_key_name(self, key, field):
+        """ get key name to use to cache any property against field name and identification key """
+        return "mobile_api.app_version_upgrade.{}.{}".format(field, key)
+
+    def _get_version_info(self, request):
+        """
+        It sets request cache data for last_supported_date and latest_version with memcached values if exists against
+        user app properties else computes the values for specific platform and sets it in both memcache (for next
+        server interaction from same app version/platform) and request cache
+        """
+        user_agent = request.META.get('HTTP_USER_AGENT')
+        if user_agent:
+            platform = self._get_platform(request, user_agent)
+            if platform:
+                request_cache_dict = request_cache.get_cache(self.REQUEST_CACHE_NAME)
+                request_cache_dict[self.USER_APP_VERSION] = platform.version
+                cached_data = cache.get_many([
+                    self._get_cache_key_name(platform.version, self.LAST_SUPPORTED_DATE_HEADER),
+                    self._get_cache_key_name(platform.NAME, self.LATEST_VERSION_HEADER)
+                ])
+
+                last_supported_date = cached_data.get(
+                    self._get_cache_key_name(platform.version, self.LAST_SUPPORTED_DATE_HEADER),
+                    None
+                )
+                if not last_supported_date:
+                    last_supported_date = self._get_last_supported_date(platform.NAME, platform.version)
+                    cache.set(
+                        self._get_cache_key_name(platform.version, self.LAST_SUPPORTED_DATE_HEADER),
+                        last_supported_date,
+                        self.MEM_CACHE_TIMEOUT
+                    )
+                request_cache_dict[self.LAST_SUPPORTED_DATE_HEADER] = last_supported_date
+
+                latest_version = cached_data.get(
+                    self._get_cache_key_name(platform.NAME, self.LATEST_VERSION_HEADER),
+                    None
+                )
+                if not latest_version:
+                    latest_version = self._get_latest_version(platform.NAME)
+                    cache.set(
+                        self._get_cache_key_name(platform.NAME, self.LATEST_VERSION_HEADER),
+                        latest_version,
+                        self.MEM_CACHE_TIMEOUT
+                    )
+                request_cache_dict[self.LATEST_VERSION_HEADER] = latest_version
+
+                return {
```

Comment:
You can just return the `request_cache_dict` instead of constructing the dict again.